src/threaded.rs

```diff
@@ -480,9 +490,34 @@ fn clipboard_thread(
                 ThreadRequest::Kill => break,
             }
         }
-        // Dispatch the event queue and block for 50 milliseconds
-        event_queue.dispatch_pending().unwrap();
-        sleep(Duration::from_millis(50));
+        // Dispatch the event queue and block for `sleep_amount`
+        let pending_events = event_queue.dispatch_pending().unwrap();
+        let num_seats = seat_map.lock().unwrap().len();
+
+        // If some app is trying to spam us when there no seats, it's likely that someone is
+        // trying to paste from us
+        if num_seats == 0 && pending_events != 0 {
```

Comment:
I'm a bit worried about this check, the idea was to provide fast paste when other app requests a text from alacritty. So I wonder whether it's good in it's this shape. It worked fine on sway, but I don't want to burn CPU occasionally.

Reply:
What's the rationale for the check `num_seats == 0` ? I'd expect this kind of test to be almost always `false`.

Reply:
@vberger
No it's not. We're creating an removing seats on the fly https://github.com/Smithay/smithay-clipboard/blob/e10bbfac70cef59c86068056e95710c39989b3d6/src/threaded.rs#L627 , so the idea behind this check is to provide speed up when you're pasting from smithay-clipboard to a different app, in this case seat map should be empty, because of KbdLeave event.

You're right that this check will be `false` in a normal use case, but this wasn't its goal, for a normal use case I'm just resetting sleep sleep_amount on each user request, so should also be fine.

If you have any thoughts on how could I differently approach the described above behavior, when we're pasting from smithay-clipboard to e.g. different alacritty instance, I would like to hear.

Reply:
Oh right, the map only contains seats that currently focus the app. My bad, I misunderstood the intent of the code.